### ISSUE TITLE
[BUG] <title>excessive overheating

### ISSUE THREAD

**Issue by Whoisipo** [session-foundation/session-ios#11]
### Code of conduct

- [X] I have read and agree to adhere to the [Code of Conduct](https://github.com/oxen-io/session-ios/blob/master/CODE_OF_CONDUCT.md)

### Self-training on how to write a bug report

- [X] I have learned [how to write a bug report](https://www.browserstack.com/guide/how-to-write-a-bug-report)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

On an ipad, for the past couple months the chat has been plagued with constant issues:  from absolute disaster when it comes to synchronization (I’ve been using a single address on iOS iPadOS and MacOS ) to the overall inability to sending or receiving files/messages send /receive simple text message .
For the past couple weeks my iPad is getting really hot when I leave the iPad on probably any live chat,. Probably waiting on some gif or an image to get through the pipe.

### Expected Behavior

To just work!

### Steps To Reproduce

Wait on some image /file to get downloaded and leave the iPad on the relevant chat/session window.

### iOS Version

iPadOS 17.5.1

### Session Version

2.6.2(452)

**Comment by naturalbornhamburger** [comment]
Hi, I am a new user, both of session and github.
Obviously my problem with session seems to fit in here.
I have serious battery usage problems with the current iOS version on my iPhone 12 Pro and I can also observe this on an iPhone 7 Plus, both with their latest iOS Versions and Session 2.8.2. It appears not to be a constant behaviour which is easily to be reproduced.
On the iPhone 12 Pro:
1h 32 minutes in foreground, 
42 minutes in background = 2h14m total takes 46% battery usage, though display brightness was set quite low and used in an inside, quite dark environment. Plus the phone heated up quite strong.
A friend with an Android phone seems to face similiar experiences.
Any ideas? Thank you.

I tried to contact session internal group but I was rejected to post my message there with the note that new users are not permitted to post there.

**Comment by KeeJef** [comment]
Hmmm it does seem like Session is a bit of a battery hog on my device too, i think the increase in battery usage coincided with the move to libquic for communication with Service Nodes, is this your understanding too @mpretty-cyro ? If so, we are working on some libquic changes, not sure if any of those changes are likely to reduce battery usage though?

**Comment by mpretty-cyro** [comment]
> Hmmm it does seem like Session is a bit of a battery hog on my device too, i think the increase in battery usage coincided with the move to libquic for communication with Service Nodes, is this your understanding too [@mpretty-cyro](https://github.com/mpretty-cyro) ? If so, we are working on some libquic changes, not sure if any of those changes are likely to reduce battery usage though?

Yep pretty much - for the worst 10% of cases the battery usage increased by:
- **While the app is in the foreground:** about 0.7% per day (from 3.6% to 4.4%)
- **While the app is in the background:** about 2.65% per day (from 0.25% to 2.9%)

It's hard to say why the background usage has increased this much compared to the foreground usage but I know I've also fixed a few bugs with background polling and finishing off work when immediately being sent to the background, so not sure if those have had an impact (eg. maybe it didn't work at all previously)

@KeeJef There may also be additional tweaks we can make when running in the background to disable some of the logic so that it takes less work (eg. refreshing the snode pool, maybe a couple of other things) but a number of the changes we could make will likely result in making and background work less reliable if the app isn't opened frequently enough 😕

**Comment by naturalbornhamburger** [comment]
My iPhone recently (yesterday) heated up so much while using Session in foreground, that it even stopped charging due to overheating. And that in an inside room with little light and at 20°C room temperature. Never had something like that before with any other app.

**Comment by naturalbornhamburger** [comment]
I‘d appreciate letting me know if there will be cure for this disease, pls. see attached two screenshots. I will have to ditch Session otherwise since it is literally destroying the iPhone battery.

![Image](https://github.com/user-attachments/assets/2f05dd2d-ec5f-448c-91cd-535aa9de3eab)

**Comment by KeeJef** [comment]
Yeah, we're merging in some libquic fixes into the GroupsV2 release which should be coming out soon. We're hoping that these fixes will resolve issues where libquic was using 100% of CPU in certain cases. We think this 100% CPU usage may be leading to increased battery drain, so resolving these issues in libquic should reduce battery drain. But we wont know this for sure until we deploy the new GroupsV2 release.

**Comment by mpretty-cyro** [comment]
@naturalbornhamburger I just wanted to give you a heads up that the `libQuic` changes won't be included in the upcoming GroupsV2 release (`2.9.0`) but are a high priority so should be included a release shortly after - the main reason for this is that `2.9.0` has already completed internal testing whereas the `libQuic` changes are still being tested

If you wanted to track the changes these are the PRs involved (the first one being the core changes and the other two just updating `libSession` to use the new code):
- https://github.com/oxen-io/oxen-libquic/pull/156
- https://github.com/oxen-io/oxen-libquic/pull/157
- https://github.com/session-foundation/libsession-util/pull/26

Once these have all been merged and the testing completed there will be a PR on this repo to use a new version of `libSession` that includes the changes

**Comment by naturalbornhamburger** [comment]
@mpretty-cyro : Thank you just for not getting me the feeling of not being forgotten. Would you mind changing your point of view into a plain user ones? Meaning that I have no clue what needs to be done, HOW LONG it takes (I started my conversation with KeeJeff back in November I think) nor do I have any ressources to estimate myself if and when it is solved at all? As I said: It affects two totally different iPhones with various iOS versions heavily, renders Session useless and not recommendable for mobile use on iPhones imho. So - what is the expected time line explained for dummies? I mean, the origin of this thread is already getting close to a year now indicating that something could be terribly wrong from a user perspective.
Thank you very much.

**Comment by mpretty-cyro** [comment]
@naturalbornhamburger we've just started the rollout for [Release 2.10.0](https://github.com/session-foundation/session-ios/pull/414) which includes a few updates to the networking code and a fix for at least one behaviour that could result in excess CPU usage (the app may not automatically update for a few days, but you can manually install the update via the AppStore)

Let us know if you still happen to experience the battery drain issues as we are also looking at adding a setting to switch between using `Quic` and `HTTPS/TCP` for sending network requests which may also help if the new fixes don't resolve the issue

**Comment by naturalbornhamburger** [comment]
Thanks a lot. I forced the update on iOS today and keep you informed, will take a little time (maybe upto a week).
Do the changes affect also Android users?

**Comment by mpretty-cyro** [comment]
> Thanks a lot. I forced the update on iOS today and keep you informed, will take a little time (maybe upto a week).

Thanks 🙏 

>  Do the changes affect also Android users?

Nah this issue only affected the iOS app (as it's the only one using the updated networking at this stage), if you know some Android users that have been experiencing battery drain then it's probably best to open an issue on the Android GitHub page so it can be properly investigated and tracked there

**Comment by naturalbornhamburger** [comment]
Sadly, I still find the result dissatisfying. The iPhone still heats up significantly during a plain text chat conversation without any file transfer. I attach again my battery usage stats in comparison with Safari, so you see the drastic difference.

![Image](https://github.com/user-attachments/assets/24777866-39fb-433b-87f8-1b2186ab2ff1)

**Comment by naturalbornhamburger** [comment]
I am too lazy to open up another ticket here. While I made the screenshots for the above postings regarding battery usage in iOS, my chat partner received a message, that I made three screenshots (in Session!). How can that be? This has to be solved please asap to avoid misunderstandings!
Thank you.

**Comment by mpretty-cyro** [comment]
Hey @naturalbornhamburger, damn I really thought we'd tracked down the issue you were being affected by - thanks for letting us know it's still a problem. It there any chance you can share your app logs? They might help us narrow down the cause - you can export them by opening Settings in Session (profile button in the top left), selecting Help and then Report a Bug.

If there is nothing in the logs then we may not be able to resolve it until we the networking rework I mentioned earlier (adding a setting to switch between `Quic` and `HTTPS/TCP`) which is probably going to be at least a month due as other priorities keep getting in the way.

> While I made the screenshots for the above postings regarding battery usage in iOS, my chat partner received a message, that I made three screenshots (in Session!). How can that be? This has to be solved please asap to avoid misunderstandings!

This is very interesting - Session has a, somewhat flawed, feature where if you take a screenshot of a conversation it notifies the other participant but this shouldn't be happening while the app isn't open. The way this feature works is we essentially ask the operating system to tell us when a screenshot is taken and, when it lets us know, we send the notification method - Apple is pretty strict about their security so continuing to notify us when the app is in the background seems like a bug on their side 😅 

I'll raise this one as an issue internally though as there has already been a number of discussions about the _actual_ usefulness of this feature

**Comment by naturalbornhamburger** [comment]
I will look into the logs and come back to it. I would not be happy to publish logfiles in the wild.

Regarding the screenshots: I indeed made ONE inside Session, the settings screen to copy and paste the version number at the bottom from the screenshot. Nevertheless - my chat partner received 3 notifications in our chat conversation window about me, making 3 screenshots. Two of them had indeed been the two I edited side by side for easier viewing as uploaded above.

**Comment by naturalbornhamburger** [comment]
Sent todays‘ log file to @KeeJef via Session chat.

**Comment by mpretty-cyro** [comment]
> Regarding the screenshots: I indeed made ONE inside Session, the settings screen to copy and paste the version number at the bottom from the screenshot. Nevertheless - my chat partner received 3 notifications in our chat conversation window about me, making 3 screenshots. Two of them had indeed been the two I edited side by side for easier viewing as uploaded above.

I've just opened PR #432 to remove the screenshot notification since it's buggy and pretty ineffective - it'll be included in a future release

> Sent todays‘ log file to @KeeJef via Session chat.

Awesome, thanks for that - I'll ask him to send it across and take a look

**Comment by naturalbornhamburger** [comment]
> Awesome, thanks for that - I'll ask him to send it across and take a look

Hi There. Did the information contained in the logs I supplied give you any hint, how to fix the heat / battery usage problem?
Thanks.

**Comment by mpretty-cyro** [comment]
> Hi There. Did the information contained in the logs I supplied give you any hint, how to fix the heat / battery usage problem? Thanks.

Hey @naturalbornhamburger, sorry I thought I'd already sent an update for this - I went through the logs and unfortunately there weren't any red flags in there

The most likely culprit is still the networking code (there is probably another infinite loop it can get stuck in somewhere), the plan is still to rework this code (which should result in fixing this problem) but unfortunately we've had some server-side work that has taken priority for the past few weeks

This should be the next thing to be worked on once that quietens down a bit but it's difficult to give any estimates at this stage about when a fix would be released unfortunately
